crypto/async/async_wait.c

```diff
@@ -113,22 +113,25 @@ int ASYNC_WAIT_CTX_get_changed_fds(ASYNC_WAIT_CTX *ctx, OSSL_ASYNC_FD *addfd,
 {
     struct fd_lookup_st *curr;
 
-    *numaddfds = ctx->numadd;
-    *numdelfds = ctx->numdel;
```

Comment:
That would also be a valid solution as long as the we check that `curr->add == 1` before actually removing the node from the list.
I agree it would be cleaner, although in terms of performance we don't expect a big difference since the first call to `ASYNC_WAIT_CTX_get_changed_fds()` is generally followed by a second one that must loop to get all the fds.
Would you prefer to go with your solution and remove the node?

I agree that the two counters are redundant in this case.